contracts/instruments/RibbonThetaVault.sol

```diff
@@ -278,6 +282,50 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
         return amountAfterFee;
     }
 
+    /**
+     * @notice Lock's users shares for future withdraw and ensures that the new short excludes the scheduled amount.
+     * @param shares is the number of shares to be withdrawn in the future.
+     */
+    function withdrawLater(uint256 shares) external nonReentrant {
+        require(shares > 0, "!shares");
+        require(
+            scheduledWithdrawals[msg.sender] == 0,
+            "Scheduled withdrawal already exists"
+        );
+
+        scheduledWithdrawals[msg.sender] = shares;
+        queuedWithdrawShares = queuedWithdrawShares.add(shares);
+        _transfer(msg.sender, address(this), shares);
+
+        emit ScheduleWithdraw(msg.sender, shares);
+    }
+
+    /**
+     * @notice Burns user's locked tokens and withdraws assets to msg.sender.
+     */
+    function completeScheduledWithdrawal() external nonReentrant {
+        uint256 withdrawShares = scheduledWithdrawals[msg.sender];
+        require(withdrawShares > 0, "Scheduled withdrawal not found");
+
+        scheduledWithdrawals[msg.sender] = 0;
+        queuedWithdrawShares = queuedWithdrawShares.sub(withdrawShares);
+
+        (uint256 amountAfterFee, uint256 feeAmount) =
+            withdrawAmountWithShares(withdrawShares);
+
+        _burn(address(this), withdrawShares);
+        IERC20(asset).safeTransfer(feeRecipient, feeAmount);
+        if (asset == WETH) {
+            IWETH(WETH).withdraw(amountAfterFee);
+            (bool success, ) = msg.sender.call{value: amountAfterFee}("");
+            require(success, "ETH transfer failed");
+        } else {
+            IERC20(asset).safeTransfer(msg.sender, amountAfterFee);
+        }
+
+        emit ScheduledWithdrawCompleted(msg.sender, amountAfterFee);
```

Comment:
One last thing missing, we also need to emit the `Withdraw` event here

Reply:
Resolved in https://github.com/ribbon-finance/structured-products/pull/128/commits/5e4f789936bf2c268fd3addef9bc68b332793762